components/com_users/Model/UserModel.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  com_users
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+namespace Joomla\Component\Users\Site\Model;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\MVC\Model\ItemModel;
+
+/**
+ * Public Profile model class for Users.
+ *
+ * @since  4.0
+ */
+class UserModel extends ItemModel
+{
+	/**
+	 * Method to auto-populate the model state.
+	 *
+	 * Note. Calling getState in this method will result in recursion.
+	 *
+	 * @return  void
+	 *
+	 * @since   1.6
+	 */
+	protected function populateState()
+	{
+		$app = \JFactory::getApplication();
+
+		// Load state from the request.
+		$pk = $app->input->getInt('id');
+		$this->setState('user.id', $pk);
+
+		$offset = $app->input->getUInt('limitstart');
+		$this->setState('list.offset', $offset);
+
+		// Load the parameters.
+		$params = $app->getParams();
+		$this->setState('params', $params);
+
+	}
+
+	/**
+	 * Method to get user data.
+	 *
+	 * @param   integer  $pk  The id of the user.
+	 *
+	 * @return  object|boolean  Menu item data object on success, boolean false
+	 *
+	 * @throws \Exception
+	 */
+	public function getItem($pk = null)
+	{
+		$pk = (!empty($pk)) ? $pk : (int) $this->getState('user.id');
+
+		if ($this->_item === null)
+		{
+			$this->_item = array();
+		}
+
+		if (!isset($this->_item[$pk]))
+		{
+			try
+			{
+				$db = $this->getDbo();
```

Comment:
Can you try it with JUser::getInstance($pk); then you have the whole user object already.